source/detail/number_format/number_formatter.cpp

```diff
@@ -1356,9 +1357,8 @@ format_color number_format_parser::color_from_string(const std::string &color)
         }
         else if ((color.substr(0, 5) == "Color") || (color.substr(0, 5) == "COLOR"))
         {
-            auto color_number = std::stoull(color.substr(5));
-
-            if (color_number >= 1 && color_number <= 56)
+            unsigned int color_number = 0;
```

Comment:
Can we change this type to `format_color`?

Reply:
Unfortunately not, no, because `format_color` is of type `enum class`, which is type-safe and thus cannot be casted implicitly to any other type (so it needs an explicit cast). Even if this was possible (e.g. by using a simple `enum` instead of an `enum class`), [according to the C++ specification](https://en.cppreference.com/w/cpp/language/enum), an `enum` can use any integral type, meaning that if `short`, `unsigned short`, `char`, `signed char` or `unsigned char` would be used, we wouldn't have parsers for these types (at least not yet, and I doubt it would be worth adding them).

Reply:
I think it would be possible to keep `enum class` and use template programming to automatically safely cast it using `std::underlying_type`. As we do not need it a lot, you may just leave it as is.